js/main.js

```diff
@@ -276,13 +280,44 @@ function updateEvents() {
 
 }
 
+function cycleEvents() {
+  for (i = 0; i < events.length; i++) {
+    if (events[i].items.length>3) {
+      if (eventsOffset[i] !== undefined) {
+        if (eventsOffset[i] >= (events[i].items.length - 3)) {
+          eventsOffset[i] = 0;
+	} else {
+          eventsOffset[i]++;
+	}
+      } else {
+        eventsOffset[i] = 0;
+      }
+    }
+  }
+  animateEventRotation();
+}
+
+function animateEventRotation() {
+  // Written for use with support for multiple .events classes to allow cycling of mulitiple days
+  for (i = 0; i < $('.events').length; i++) {
+    if (eventsOffset[i] > 0) {
+      for (j = 0; j < eventsOffset[i]; j++) {
+        $('.events:eq('+i+') ul li:eq('+j+')').hide("slow");
+      }
+    } else {
+      $('.events:eq('+i+') ul li').show("slow");
+    }
+  }
+}
+
 $(document).ready(function() {
     updateEvents();
     updateDayMode();
 
     setInterval(updateTime, 1000);
     setInterval(updateDate, 1000);
     setInterval(switchLocale, 5000);
+    setInterval(cycleEvents, 7000);
```

Comment:
No idea how often the shuffling of events should happen.

I just figured it might be a bit weird if the shuffling happens at the same time as language change? 

Not sure, interested to see how often others think the shuffling should happen.